pkg/controller/directimagemigration/imagestream.go

```diff
@@ -0,0 +1,120 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package directimagemigration
+
+import (
+	"context"
+	//  liberr "github.com/konveyor/controller/pkg/error"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	imagev1 "github.com/openshift/api/image/v1"
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) getDirectImageStreamMigrations() ([]migapi.DirectImageStreamMigration, error) {
+	ismList := migapi.DirectImageStreamMigrationList{}
+	err := t.Client.List(
+		context.TODO(),
+		k8sclient.MatchingLabels(t.Owner.GetCorrelationLabels()),
+		&ismList)
+	if err != nil {
+		return nil, err
+	}
+	return ismList.Items, nil
+}
+
+func (t *Task) createDirectImageStreamMigrations() error {
+	// Get client for source
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+	// Get list namespaces to iterate over
+	for srcNsName, destNsName := range t.Owner.GetNamespaceMapping() {
+		isList := imagev1.ImageStreamList{}
+		err := srcClient.List(
+			context.TODO(),
+			k8sclient.InNamespace(srcNsName),
+			&isList)
+		if err != nil {
+			return err
+		}
+		for _, is := range isList.Items {
+			ismList := migapi.DirectImageStreamMigrationList{}
+			err := t.Client.List(
+				context.TODO(),
+				k8sclient.MatchingLabels(t.Owner.DirectImageStreamMigrationLabels(is)),
+				&ismList)
+			if err != nil {
+				return err
+			}
+			if len(ismList.Items) > 0 {
+				continue
+			}
+			imageStreamMigration := t.buildDirectImageStreamMigration(is, destNsName)
+			err = t.Client.Create(context.TODO(), &imageStreamMigration)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+func (t *Task) buildDirectImageStreamMigration(is imagev1.ImageStream, destNsName string) migapi.DirectImageStreamMigration {
+	labels := t.Owner.DirectImageStreamMigrationLabels(is)
+	imageStreamMigration := migapi.DirectImageStreamMigration{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:       labels,
+			GenerateName: t.Owner.GetName() + "-",
+			Namespace:    t.Owner.Namespace,
+		},
+		Spec: migapi.DirectImageStreamMigrationSpec{
+			SrcMigClusterRef:  t.Owner.Spec.SrcMigClusterRef,
+			DestMigClusterRef: t.Owner.Spec.DestMigClusterRef,
+			ImageStreamRef: &kapi.ObjectReference{
+				Name:      is.Name,
+				Namespace: is.Namespace,
+			},
+		},
+	}
+	t.setDirectImageStreamOwnerReference(&imageStreamMigration)
+	if is.Namespace != destNsName {
+		imageStreamMigration.Spec.DestNamespace = destNsName
+	}
+	return imageStreamMigration
+}
+
+func (t *Task) setDirectImageStreamOwnerReference(dism *migapi.DirectImageStreamMigration) {
+	for i := range dism.OwnerReferences {
+		ref := &dism.OwnerReferences[i]
+		if ref.Kind == t.Owner.Kind {
+			ref.APIVersion = t.Owner.APIVersion
+			ref.Name = t.Owner.Name
+			ref.UID = t.Owner.UID
+			return
+		}
+	}
+	dism.OwnerReferences = append(
```

Comment:
I wonder if there should be more owner references than just the t.Owner?

We could replace this with

```
dism.OwnerReferences = []metav1.OwnerReference{
...
...
}
```

A list with just one entry

Reply:
Yeah, I guess this is attempting to be more generic than is really necessary (taken from the migmigration controller). It's probably simpler to just assign a single one here.